dashboard/src/views/Projects/Versions/index.tsx

```diff
@@ -1,5 +1,143 @@
+import SearchIcon from "@mui/icons-material/Search";
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  Sheet,
+  Table,
+  Typography,
+} from "@mui/joy";
+
+import { useEffect, useState } from "react";
+import { mockedVersionsResponse } from "../../../api/mocks/versions.mock";
+import { PaginationSection } from "../../../components/TablePaginationSection";
+import { TableRow } from "../../../components/TableRow";
+import { GetVersionsResponse } from "../../../interfaces/versions";
+
 export const VersionsOverview = () => {
-  return <div>VersionsOverview</div>;
-};
+  const [searchBar, setSearchBar] = useState("");
+  //TODO: replace mocked data with the response from the API when react query hooks are implemented
+  const versions = mockedVersionsResponse.versions;
+  const [versionsList, setVersionsList] = useState<GetVersionsResponse>(
+    mockedVersionsResponse,
+  );
+
+  useEffect(() => {
+    if (searchBar === "") {
+      return;
+    }
+    setVersionsList(versionsList);
+  }, [versionsList, searchBar]);
+
+  const versionSearchHandle = (search: string) => {
+    const filteredVersions = versions.filter((version) =>
+      version.name.toLowerCase().includes(search.toLowerCase()),
+    );
+    setVersionsList({ versions: filteredVersions });
+  };
+
+  return (
+    <>
+      <Box
+        sx={{
+          borderRadius: "sm",
+          py: 0.5,
+          display: { xs: "none", sm: "flex" },
+          flexWrap: "wrap",
+          gap: 1.5,
+          "& > *": {
+            minWidth: { xs: "120px", md: "160px" },
+          },
+        }}
+      >
+        <Typography
+          level="h2"
+          component="h1"
+          style={{
+            alignSelf: "center",
+            fontSize: "3rem",
+            marginRight: "1.5rem",
+          }}
+        >
+          Versions
+        </Typography>
 
+        <FormControl sx={{ flex: 1, pb: "1.1rem" }} size="sm">
+          <FormLabel>Search for version</FormLabel>
 
+          <Input
+            size="sm"
+            placeholder="Look up for version name"
+            startDecorator={<SearchIcon />}
+            onChange={(e) => {
+              setSearchBar(e.target.value);
+              versionSearchHandle(e.target.value);
+            }}
+            value={searchBar}
+          />
+        </FormControl>
+      </Box>
+
+      <Button sx={{ mb: "0.5rem" }}>Add New Version</Button>
+
+      <Sheet
+        variant="outlined"
+        sx={{
+          display: { xs: "none", sm: "initial" },
+          width: "100%",
+          borderRadius: "sm",
+          flexShrink: 1,
+          overflow: "auto",
+          minHeight: 0,
```

Comment:
move to its own css file

Reply:
then we need to make css files for every component to be aligned. Not sure if it worth it for that small internal project. What would you recommend @naueramant ?

Reply:
even if it is an internal project we should follow the guidlines for project structure.